conmon/conmon.c

```diff
@@ -116,6 +116,65 @@ int set_k8s_timestamp(char *buf, ssize_t buflen, const char *stream_type)
 	return 0;
 }
 
+
+/*
+ * splits buf into lines and inserts timestamps at the beginning of
+ * the line written to logfd.
+ */
+int write_with_timestamps(int logfd, const char *buf, ssize_t buflen)
+{
+	#define TSBUFLEN 34
+	char tsbuf[TSBUFLEN];
+	static bool last_buf_ended_with_newline = TRUE;
+
+	g_auto(GStrv) lines = g_strsplit(buf, "\n", -1);
+	ssize_t num_lines = g_strv_length(lines);
+	for (ssize_t i = 0; i < num_lines; i++)
+	{
+		const char *line = lines[i];
+
+		ninfo("Processing line: %ld, %s", i, line);
+
+		/* Skip last line if it is empty */
+		if (i == (num_lines - 1) && buf[buflen-1] == '\n' && !strcmp("", line)) {
+			ninfo("Skipping last line");
+			break;
+		}
+
+		/*
+		 * Only add timestamps for first line if last buffer's last
+		 * line ended with newline. Add it for all other lines.
+		 */
+		if (i != 0 || (i == 0 && last_buf_ended_with_newline)) {
+			ninfo("Adding timestamp");
+			int rc = set_k8s_timestamp(tsbuf, TSBUFLEN, "stdout");
```

Comment:
We are taking care of partial lines in the PR. The logic will always have edge cases. For e.g. if we append after \n instead of inserting at beginning we still need to add one at the beginning of the log buffer. In my approach I am tracking if previous buffer ended with \n or not and adding a timestamp accordingly. One thing that is missing and needs to be handled is when there are no \n in a buf.